test/livebook/runtime/node_pool_test.exs

```diff
@@ -0,0 +1,91 @@
+defmodule Livebook.Runtime.NodePoolTest do
+  use ExUnit.Case, async: true
+
+  alias Livebook.Runtime.NodePool
+
+  # Tests for Livebook.Runtime.NodePool
+  #
+  # Note:
+  #   We do not spawn actual nodes as it can be time
+  #   intensive (on low spec machines)
+
+  describe "start_link" do
+    test "correctly starts a registered GenServer" do
+      start_supervised!({NodePool, name: Foo})
+
+      # Verify Process is running
+      assert Process.whereis(Foo)
+    end
+  end
+
+  describe "get_name/2" do
+    test "creates a new node name if pool is empty" do
+      start_supervised!({NodePool, name: Bar})
+
+      # Assert that we get a result and that it is an atom
+      result = NodePool.get_name(Bar, node())
+      assert result
+      assert is_atom(result)
+    end
+
+    test "adds a new node name to given pool when generated" do
+      start_supervised!({NodePool, name: Baz})
+
+      result = NodePool.get_name(Baz, node())
+
+      assert generated_contains?(Baz, result)
+    end
+
+    test "returns an existing name if pool is not empty" do
+      start_supervised!({NodePool, name: Fu})
+
+      name = NodePool.get_name(Fu, node())
+      send(Fu, {:add_node, name})
```

Comment:
Another idea would be to set the time buffer to 1ms and send an actual `:nodedown` message here, so that it better mimics the reality (you may just need `Process.sleep(1)` afterwards, but I that should be fine.

Reply:
@jonatanklosko , That's what I was doing in the first commit, but if you look [here](https://github.com/Benjamin-Philip/livebook/blob/9f7668bc821cff6e8b8aeec74fbe9d83be91673d/lib/livebook/runtime/node_pool.ex#L64), our timer sends that message to `__MODULE__`, so it wouldn't work in differently named/registered instances. I could write another function to handle the afore mentioned case, but it just feels like we are just adding more complexity for tests.

Reply:
Ah, that should actually be `Process.send_after(self(), {:add_node, node}, state.buffer_time)`, so that we send the message to ourselves regardless the registered name. Sorry I didn't notice this earlier!